docs/BUILD.bazel

```diff
@@ -2,49 +2,72 @@ load("@bazel_skylib//rules:write_file.bzl", "write_file")
 load("@bazel_skylib//rules:diff_test.bzl", "diff_test")
 load("@io_bazel_stardoc//stardoc:stardoc.bzl", "stardoc")
 
+# Dictionary of key-value pairs where:
+# key := {bzl_library target}
+# value := ({output MD file}, {rule template}, {footer file})
 _DOC_SRCS = {
-    "//extras:embed_data": "extras.md",
+    "//extras:embed_data": ("extras.md", "rule_subsection.vm", "footers/empty.md"),
```

Comment:
why parameterize the second value if it's constant? let's keep this as simple as possible.

if you're going to have footers, just make one for each file, like `footers/extras.md` and `footers/core.md` 

then the configuration can just be a string -> string dictionary.

ALTERNATIVELY

If the configuration dictionary gets unwieldy, and has only two entries, that's a sign that it's a bad abstraction. Just stamp out two copies of each rule below, it's syntactically simpler and duplication in a BUILD file is less of a smell than it would be in source code

Reply:
Actually good thing you pointed that out, I meant to use different templates for both. 

I like the idea of making the value another dictionary tho, I'll update. 

The reason why I created a placeholder footer file called `empty` is because some docs won't need one, but the genrule needs some file there in `srcs` for each entry in _DOC_SRCS.